### PR TITLE
Lab3 - implemented letter count also sorting, tests included - alfredxiao

### LINKED ISSUE #415: Build lab3 code, implement letter count as well as sorting

### PR DESCRIPTION
**Pull request by alfredxiao**
Fixes #415
Review colleague's PR #318 

Proposed Changes to this PR:
- count character occurrences in string
- sort the counting alphabetically

### CODE REVIEW

**Comment by davidbroughsmyth** [comment]
As per course notes, you need to add the label "Ready for review" and "lab3" if you want this reviewed

File: 03_letters/alfredxiao/main.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+func letters(s string) map[rune]int {
+	counts := make(map[rune]int)
+	for _, r := range s {
+		counts[r]++
+	}
+	return counts
+}
+
+func sortLetters(m map[rune]int) []string {
+	rcs := []string{}
```
Comment (juliaogris) [inline 03_letters/alfredxiao/main.go:22]: nit `make([]string, 0, len(m))` - you already know the capacity.

File: 03_letters/alfredxiao/main_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	assert.Equal(t, map[rune]int{'a': 1}, letters("a"))
+	assert.Equal(t, map[rune]int{'a': 3}, letters("aaa"))
+	assert.Equal(t, map[rune]int{' ': 1}, letters(" "))
+	assert.Equal(t, map[rune]int{' ': 3}, letters("   "))
+}
+
+func TestLetterCountsUnicode(t *testing.T) {
+	assert.Equal(t, map[rune]int{'\u2318': 1}, letters("\u2318"))
+	assert.Equal(t, map[rune]int{'\u2318': 1, '\u2319': 1}, letters("\u2318\u2319"))
+	assert.Equal(t, map[rune]int{'\u2318': 1, '\u2319': 1, 'A': 1}, letters("\u2318A\u2319"))
```
Comment (juliaogris) [inline 03_letters/alfredxiao/main_test.go:21]: Nit: Could you add something like this as well please?
```go
assert.Equal(t, expected, letters("😃😃😃😃"))
```

**Review by juliaogris — APPROVED** [review]
Just a couple of minor nit picks.
You can merge if you don't want to change any more.

**Pull request merged**